rfcs/0038-cheque/0038-cheque.md

```diff
@@ -0,0 +1,324 @@
+
+```
+
+Here the transaction inputs contain a receiver secp256k1_blake160 cell whose first 20 bytes of lock script hash is equal to the `receiver_secp256k1_blake2b_lock_hash[0..20]` of the cheque cell lock args, and the signature can be validated by the receiver public key hash, and the sum of sender output cells capacity is equal to the sum of the cheque input cells capacity, the cheque cell can be unlocked.
+
+### Withdraw
+
+#### 1. Withdraw via sender signature
+
+```
+Inputs:
+    Cheque Cell:
+        Capacity: 165 CKBytes
+        Lock:
+            code_hash: <cheque_cell_script_code_hash>
+            args: <receiver_secp256k1_blake2b_lock_hash[0..20]> <sender_secp256k1_blake2b_lock_hash[0..20]>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+        Since: 0xA000000000000006
+
+Outputs:
+    SUDT Cell:
+        Capacity: 164.99 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+
+Witnesses:
+    < valid signature for sender public key hash >
+```
+
+When a signature is provided and can be validated by the sender public key hash, and the since of the cheque input cell is same as `0xA000000000000006` which means the tx failed verification unless it is 6 epochs later since the input cells get confirmed on-chain, the cheque cell can be unlocked. In this example a cheque cell is converted back to a sender normal cell with SUDT.
+
+#### 2. Withdraw via sender lock script
+
+```
+Inputs:
+    Cheque Cell:
+        Capacity: 165 CKBytes
+        Lock:
+            code_hash: <cheque_cell_script_code_hash>
+            args: <receiver_secp256k1_blake2b_lock_hash[0..20]> <sender_secp256k1_blake2b_lock_hash[0..20]>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+        Since: 0xA000000000000006
+    Normal Cell:
+        Capacity: 200 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+
+Outputs:
+    SUDT Cell:
+        Capacity: 165 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+        Type: <sudt_type_script>
+        Data:
+            sudt_amount: 100 UDT
+    Normal Cell:
+        Capacity: 199.99 CKBytes
+        Lock: <sender_secp256k1_blake2b_lock_script>
+
+Witnesses:
+    < 0x >
+    < valid signature for sender public key hash >
+```
+
+Here the transaction inputs contain a sender secp256k1_blake160 cell whose first 20 bytes of lock script hash is equal to the `sender_secp256k1_blake2b_lock_hash[0..20]` of the cheque cell lock args, and the signature can be validated by the sender public key hash, and the since of the cheque input cell is same as `0xA000000000000006` which means the tx failed verification unless it is 6 epochs later since the input cells get confirmed on-chain, the cheque cell can be unlocked. In this example a cheque cell is converted back to a sender normal cell with SUDT.
```

Comment:
```suggestion
In this example, the transaction inputs include a sender secp256k1_blake160 cell whose first 20 bytes of lock script hash is equal to the `sender_secp256k1_blake2b_lock_hash[0..20]` of the cheque cell lock args, and the signature can be validated by the sender public key hash, and the since of the cheque input cell is same as `0xA000000000000006` which means the tx failed verification unless it is 6 epochs later since the input cells get confirmed on-chain, the cheque cell can be unlocked. In this example a cheque cell is converted back to a sender normal cell with SUDT.
```